template.tpl

```diff
@@ -1071,6 +1080,48 @@ ___TEMPLATE_PARAMETERS___
       }
     ]
   },
+  {
+    "type": "GROUP",
+    "name": "buttonClickTrackingConfig",
+    "displayName": "Filter",
+    "groupStyle": "ZIPPY_OPEN",
+    "subParams": [
+      {
+        "type": "SELECT",
+        "name": "buttonClickTrackingFilterChoice",
+        "displayName": "Filter Type",
+        "macrosInSelect": false,
+        "selectItems": [
+          {
+            "value": "allowlist",
+            "displayValue": "Allowlist"
+          },
+          {
+            "value": "denylist",
+            "displayValue": "Denylist"
+          }
+        ],
+        "simpleValueType": true,
+        "subParams": [
+          {
+            "type": "TEXT",
+            "name": "buttonClickTrackingFilters",
+            "displayName": "Classes",
+            "simpleValueType": true,
+            "help": "A comma-seperated list of HTML class names for filtering.",
+            "valueHint": "htmlClass, secondHtmlClass"
```

Comment:
Does this need the "Allow empty strings"/`canBeEmptyString` option to not be `undefined` in the code later and cause a JS error?

Reply:
I've handled `undefined`/general falsey in the JS, `undefined` is treated as no filter options being set for the button tracking now.